lib/iterator.js

```diff
@@ -199,6 +199,68 @@ class Iterator {
     return result + suffix;
   }
 
+  // Find value in this iterator compared by `comparator` on values
+  // obtained by applying `accessor` to each value
+  // Signature: comparator[, accessor[, thisArg]]
+  //   comparator <Function> returns true if new value should be accepted
+  //     currValue <any> current value, starts with undefined
+  //     nextValue <any> next value
+  //     Returns: <boolean> true if next value should be accepted
+  //   accessor <Function> gets value to compare by, current iterator value
+  //       is used by default
+  //     value <any> current iterator value
+  //     Returns: <any> value to compare by
+  //   thisArg <any> value to be used as `this` when calling `accessor` and
+  //       `comparator`
+  // Returns: last iterator value where `comparator` returned `true`,
+  //     `undefined` by default
+  findCompare(comparator, accessor, thisArg) {
+    let res = undefined;
+    let resCompareBy = undefined;
+    for (const value of this) {
+      const compareBy = accessor ? accessor.call(thisArg, value) : value;
+      if (comparator.call(thisArg, resCompareBy, compareBy)) {
+        resCompareBy = compareBy;
+        res = value;
+      }
+    }
+    return res;
+  }
+
+  // Find maximum value in this iterator compared by values obtained by
+  // applying `accessor` to each value
```

Comment:
Just to simplify the description:
```suggestion
  // Find the maximum value in this iterator
```